### PR TITLE
Initialize client id if need before anything else (#976)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
Fix issue #976

### CODE REVIEW

**Review by dinhlongviolin1 — APPROVED** [review]
🥇 🥇 🥇

File: src/taipy/gui/gui.py [hunk 1]
```diff
@@ -460,9 +460,12 @@ def _get_client_id(self) -> str:
             else getattr(g, Gui.__ARG_CLIENT_ID, "unknown id")
         )
 
-    def __set_client_id_in_context(self, client_id: t.Optional[str] = None):
+    def __set_client_id_in_context(self, client_id: t.Optional[str] = None, force=False):
         if not client_id and request:
             client_id = request.args.get(Gui.__ARG_CLIENT_ID, "")
+        if not client_id and force:
+            res = self._bindings()._get_or_create_scope("")
```
Comment (dinhlongviolin1) [inline src/taipy/gui/gui.py:467]: Instead of an empty string, should this be a defined value

File: src/taipy/gui/server.py [hunk 2]
```diff
@@ -102,8 +102,9 @@ def __init__(
         self._is_running = False
 
         # Websocket (handle json message)
+        # adding args for the one call with a server ack request
         @self._ws.on("message")
-        def handle_message(message) -> None:
+        def handle_message(message, *args) -> None:
```
Comment (dinhlongviolin1) [inline src/taipy/gui/server.py:107]: I dont see that the args is being passed into any other function. Am I missing something here

File: src/taipy/gui/gui.py [hunk 3]
```diff
@@ -954,7 +959,8 @@ def _send_ws_id(self, id: str) -> None:
             {
                 "type": _WsType.CLIENT_ID.value,
                 "id": id,
-            }
+            },
+            allow_grouping=False,
```
Comment (dinhlongviolin1) [inline src/taipy/gui/gui.py:963]: Nice one!

**Review by dinhlongviolin1 — APPROVED** [review]

File: src/taipy/gui/server.py (hunk 2 above)
Comment (FredLL-Avaiga) [reply src/taipy/gui/server.py:107]: I use the Ack server callback and it complained once that there were 2 arguments

File: src/taipy/gui/gui.py (hunk 1 above)
Comment (FredLL-Avaiga) [reply src/taipy/gui/gui.py:467]: nope
force is set only when called from _init_route
- if there's a client id, we won't go through this
- else we want a new id

Which should never happen as we're waiting a return before showing the page for the first time

Comment (dinhlongviolin1) [reply src/taipy/gui/gui.py:467]: thanks. Make sense to me

**Review by FabienLelaquais — APPROVED** [review]
Excellent pick!

**Pull request merged**